pkg/config/validation/validation.go

```diff
@@ -214,6 +214,21 @@ func ValidateHTTPHeaderName(name string) error {
 	return nil
 }
 
+// ValidateStringMatch validates a StringMatch
+func ValidateStringMatch(m *networking.StringMatch) error {
+	if m == nil {
+		return nil
+	}
+	switch x := m.MatchType.(type) {
+	case *networking.StringMatch_Regex:
+		// Default max size for safe regex is 100
+		if len(x.Regex) > 100 {
```

Comment:
Should we also be validating that the input is a well-formed regex?

@sushicw @davidebbo Introducing more validations in this code implicitly means that old configurations already deployed are prone to now being invalid. Seems like an area ripe for additional analysis in the config analyzer.